### PR TITLE
Dynamodb fixes2

### PR DESCRIPTION
**Pull request by niels-qup**

### CODE REVIEW

File: database/dynamo/reflect.go [hunk 1]
```diff
@@ -23,8 +23,19 @@ const (
 	dynamoNumeric  = "N"
 )
 
+var (
+	// Not thread safe but doesn't have to be.
+	definitions = make(map[string]*DynamoTableDefinition)
```
Comment (daniel-qup) [inline database/dynamo/reflect.go:28]: You can use map[reflect.Type]*DynamoTableDefinition, this makes it clearer what the map is meant to do.
Also, all the libraries that I've seen that cache computations based on reflection use the type as index directly.
I don't think there's a performance hit by using reflect.Type, but I'm not 100% sure.

Comment (niels-qup) [reply database/dynamo/reflect.go:28]: Don't agree, using the reflect.Type as map index has a performance impact of making it at-least 5x slower.

**Review by Keidjan1 — APPROVED** [review]

File: database/dynamo/reflect.go (hunk 1 above)
Comment (pcittadini) [reply database/dynamo/reflect.go:28]: Agree with Niels, invoking reflection to determine index type might only make things slower. If you know the type, just use that type :)

**Review by pcittadini — APPROVED** [review]
Won't question any choice so far, I will use this library in fulfillment in order to perform a QA.
LGTM !

**Comment by niels-qup** [comment]
I'll include this code in 1.0.0-alpha3

**Pull request merged**